src/main/scala/stuff/products.scala

```diff
@@ -1,6 +1,90 @@
 package ohnosequences.stuff
 
 import functions._
+import products._
+
+abstract
+class Product {
+
+  type On <: Category
+  val  on  : Category.is[On]
+
+  @infix
+  type ×[X <: On#Objects, Y <: On#Objects] <: On#Objects
+
+  type ∗ <: On#Objects
+
+  def both[
+    X <: On#Objects,
+    A <: On#Objects,
+    B <: On#Objects,
+  ]
+  // TODO find a convenient aliasing convention for both ×'s
+  : ohnosequences.stuff.×[On#C[X, A], On#C[X, B]] -> On#C[X, A × B]
+
+  def erase[X <: On#Objects]: On#C[X, ∗]
+
+  def  left[A <: On#Objects, B <: On#Objects]: On#C[A × B, A]
+  def right[A <: On#Objects, B <: On#Objects]: On#C[A × B, B]
+}
```

Comment:
These `left` and `right` methods are just the two projections?

Reply:
Yes!